app/src/main/java/com/github/epfl/meili/PhotoDemoActivity.kt

```diff
@@ -0,0 +1,80 @@
+package com.github.epfl.meili
+
+import android.graphics.ImageDecoder.createSource
+import android.graphics.ImageDecoder.decodeBitmap
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import android.provider.MediaStore.Images.Media.getBitmap
+import android.view.View
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import com.github.epfl.meili.storage.FirebaseStorageService
+import com.github.epfl.meili.storage.StorageService
+import com.squareup.picasso.Picasso
+
+class PhotoDemoActivity : AppCompatActivity() {
+
+    companion object {
+        var storageService: () -> StorageService = { FirebaseStorageService }
+    }
+
+    private lateinit var choose: Button
+    private lateinit var upload: Button
+    private lateinit var show: Button
+    private lateinit var imageView: ImageView
+    private var filePath: Uri? = null
+
+    /*
+        the getBitmap function has been deprecated and replaced by decodeBitmap in Android.
+        However, SDK versions < 28 don't support decodeBitmap.
+        Since Meili's min SDK version is 23, we need to support both old and new versions.
+    */
```

Comment:
Nitpick: move this comment inside of the method because like this it looks like documentation to me

Reply:
Fixed